lib/rubocop/cop/naming/method_name.rb

```diff
@@ -106,11 +106,16 @@ class MethodName < Base
         # @!method new_struct?(node)
         def_node_matcher :new_struct?, '(send (const {nil? cbase} :Struct) :new ...)'
 
+        # @!method new_data?(node)
+        def_node_matcher :new_data?, '(send (const {nil? cbase} :Data) :define ...)'
```

Comment:
Perhaps `define_data` is more appropriate than `new_data`?

```suggestion
        # @!method define_data?(node)
        def_node_matcher :define_data?, '(send (const {nil? cbase} :Data) :define ...)'
```